webapp/graphite/finders/standard.py

```diff
@@ -52,10 +53,18 @@ def find_nodes(self, query):
 
         elif isfile(absolute_path):
           if absolute_path.endswith('.wsp') and WhisperReader.supported:
+            if found_leafs >= settings.METRICS_FIND_WHISPER_TRUNCATE_THRESHOLD:
```

Comment:
you can yield later to factorize the code:

node = None
if whsiper:
  node = LeafNode(...)
elif gzip:
  node = LeafNode(...)

if node is not None:
   threshold checks
   yield node